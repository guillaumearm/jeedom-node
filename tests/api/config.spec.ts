import { prepareJeedomTest } from '../utils'

describe('rpc.config', () => {
  const [api, recordTest] = prepareJeedomTest()

  recordTest('config::byKey', async () => {
    expect(await api.config.byKey({ key: 'name' })).toMatchSnapshot()
  })

  recordTest('config::save', async () => {
    const res1 = await api.config.save({ plugin: '__test__', key: '__test__', value: 'test' })
    const value = await api.config.byKey({ plugin: '__test__', key: '__test__' })
    const res2 = await api.config.save({ plugin: '__test__', key: '__test__', value: '' })
    expect(value).toBe('test')
    expect([res1, value, res2]).toMatchSnapshot()
  })
})
