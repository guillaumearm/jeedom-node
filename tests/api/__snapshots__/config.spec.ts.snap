// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rpc.config config::byKey 1`] = `"Jeedom"`;

exports[`rpc.config config::save 1`] = `
Array [
  null,
  "test",
  null,
]
`;
